src/realm-settings/LocalizationTab.tsx

```diff
@@ -76,31 +100,159 @@ export const LocalizationTab = ({
+            },
+          },
+        ],
+      })
+    );
+    setTableRows(updatedRows);
+  }, [messageBundles]);
+
+  useFetch(
+    async () => {
+      const params = {
+        first,
+        max,
+      };
+      let result = await adminClient.realms.getRealmLocalizationTexts({
+        ...params,
         realm: realm.realm!,
         selectedLocale:
           selectMenuLocale || getValues("defaultLocale") || DEFAULT_LOCALE,
       });
 
-      return Object.entries(result);
-    } catch (error) {
-      return [];
-    }
+      const searchInBundles = (idx: number) => {
+        return Object.entries(result).filter((i) => i[idx].includes(filter));
+      };
+
+      if (filter) {
+        const filtered = uniqWith(
+          searchInBundles(0).concat(searchInBundles(1)),
+          isEqual
+        );
+
+        result = Object.fromEntries(filtered);
+      }
+
+      return { result };
+    },
+    ({ result }) => {
+      setMessageBundles(Object.entries(result).slice(first, first + max + 1));
+      return Object.entries(result).slice(first, first + max + 1);
+    },
+    [tableKey, filter, first, max]
+  );
+
+  const handleTextInputChange = (
+    newValue: string,
+    evt: any,
+    rowIndex: number,
+    cellIndex: number
+  ) => {
+    setTableRows((prev) => {
+      const newRows = cloneDeep(prev);
+      newRows[rowIndex].cells[cellIndex].props.editableValue = newValue;
+      return newRows;
+    });
   };
 
-  const tableLoader = async () => {
-    try {
-      const result = await adminClient.realms.getRealmLocalizationTexts({
-        realm: currentRealm,
-        selectedLocale: selectMenuLocale,
-      });
+  const updateEditableRows = (
+    evt: any,
+    type: RowEditType,
+    isEditable?: boolean | undefined,
+    rowIndex?: number,
+    validationErrors?: RowErrors
+  ) => {
+    if (rowIndex === undefined) {
+      return;
+    }
+    const newRows = cloneDeep(tableRows);
+    let newRow;
+    const invalid = validationErrors && Object.keys(validationErrors).length;
+    if (invalid) {
+      newRow = validateCellEdits(newRows[rowIndex], type, validationErrors);
+    } else if (type === RowEditAction.cancel) {
+      newRow = cancelCellEdits(newRows[rowIndex]);
+    } else {
+      newRow = applyCellEdits(newRows[rowIndex], type);
+    }
+    newRows[rowIndex] = newRow;
 
-      return Object.entries(result);
-    } catch (error) {
-      return [];
+    // Update the copy of the retrieved data set so we can save it when the user saves changes
+    if (!invalid && type === RowEditAction.save) {
+      const key = (newRow.cells?.[0] as IRowCell).props.value;
+      const value = (newRow.cells?.[1] as IRowCell).props.value;
+
+      // We only have one editable value, otherwise we'd need to save each
+      try {
+        adminClient.setConfig({
+          requestConfig: { headers: { "Content-Type": "text/plain" } },
+        });
+        adminClient.realms.addLocalization(
```

Comment:
This API call needs to be awaited.
```suggestion
        await adminClient.realms.addLocalization(
```